backend/MangaMagnet.Api/Service/BroadcastProgressService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Buffers;
+using System.Text.Json;
+using MangaMagnet.Core.Progress;
+using MangaMagnet.Core.Util;
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.Options;
+
+namespace MangaMagnet.Api.Service;
+
+public class BroadcastProgressService(ProgressService progressService, WebSocketService webSocketService, IOptions<JsonOptions> jsonOptions) : BackgroundService
+{
+	protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+	{
+		try
+		{
+			var timer = new PeriodicTimer(TimeSpan.FromMilliseconds(500));
+			var buffer = new PooledArrayBufferWriter<byte>();
+
+			while (await timer.WaitForNextTickAsync(stoppingToken))
+			{
+				var changedTasks = progressService.GetUpdatedTasksAndReset();
+
+				if (changedTasks.Count == 0)
+				{
+					continue;
+				}
+
+				var writer = new Utf8JsonWriter(buffer);
```

Comment:
Need to add a `try..catch` from here til the `SendToAllAsync`, so the BackgroundService doesn't quit when a client or serializer throws an error.